core/src/main/java/org/infinispan/configuration/cache/IndexMergeConfigurationBuilder.java

```diff
@@ -0,0 +1,167 @@
+package org.infinispan.configuration.cache;
+
+
+import static org.infinispan.configuration.cache.IndexMergeConfiguration.CALIBRATE_BY_DELETES;
+import static org.infinispan.configuration.cache.IndexMergeConfiguration.FACTOR;
+import static org.infinispan.configuration.cache.IndexMergeConfiguration.MAX_DOCS;
+import static org.infinispan.configuration.cache.IndexMergeConfiguration.MAX_FORCED_SIZE;
+import static org.infinispan.configuration.cache.IndexMergeConfiguration.MAX_SIZE;
+import static org.infinispan.configuration.cache.IndexMergeConfiguration.MIN_SIZE;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.infinispan.commons.configuration.Builder;
+import org.infinispan.commons.configuration.ConfigurationBuilderInfo;
+import org.infinispan.commons.configuration.attributes.Attribute;
+import org.infinispan.commons.configuration.attributes.AttributeSet;
+import org.infinispan.commons.configuration.elements.ElementDefinition;
+import org.infinispan.configuration.global.GlobalConfiguration;
+
+/**
+ * @since 12.0
+ */
+public class IndexMergeConfigurationBuilder extends AbstractIndexingConfigurationChildBuilder
+      implements Builder<IndexMergeConfiguration>, ConfigurationBuilderInfo {
+
+   private static final String KEY_PREFIX = "hibernate.search.backend.io.merge";
+   private static final String MAX_DOCS_KEY = KEY_PREFIX + ".max_docs";
+   private static final String FACTOR_KEY = KEY_PREFIX + ".factor";
+   private static final String MIN_SIZE_KEY = KEY_PREFIX + ".min_size";
+   private static final String MAX_SIZE_KEY = KEY_PREFIX + ".max_size";
+   private static final String MAX_FORCED_SIZE_KEY = KEY_PREFIX + ".max_forced_size";
+   private static final String CALIBRATE_BY_DELETES_KEY = KEY_PREFIX + ".calibrate_by_deletes";
```

Comment:
FWIW, constants are available in Hibernate Search: see `org.hibernate.search.backend.lucene.cfg.LuceneIndexSettings`.
Similarly, for backend-level settings: `org.hibernate.search.backend.lucene.cfg.LuceneBackendSettings`.

Reply:
Unfortunately I can't use them here, as ```core``` does not depend on Hibernate Search